ShowBuildMenu.bat

```diff
@@ -29,117 +30,102 @@ echo --- Exit ---
 echo X. Make the beautiful build menu go away.
 echo.
 
-%BUILDTOOL% prompt BCDEFGHIX
-if errorlevel 8 goto end
-if errorlevel 7 goto teamcity-menu
-if errorlevel 6 goto build-async
-if errorlevel 5 goto build-release-package
-if errorlevel 4 goto build-release
-if errorlevel 3 goto build-debug
-if errorlevel 2 goto test-run
+%BUILDTOOL% prompt ABCDEFGHIX
+if errorlevel 9 goto end
+if errorlevel 8 goto teamcity-menu
+if errorlevel 7 goto build-async
+if errorlevel 6 goto build-release-package
+if errorlevel 5 goto build-release
+if errorlevel 4 goto build-debug
+if errorlevel 3 goto test-run-64
+if errorlevel 2 goto test-run-32
 if errorlevel 1 goto test-activate
 if errorlevel 0 goto test-setup-menu
 
 :test-setup-menu
 echo A. Add a test configuration for SQL Server.
-echo B. Add a test configuration for Firebird (x86).
-echo C. Add a test configuration for Firebird (x64). [not recommended]
-echo D. Add a test configuration for SQLite (x86).
-echo E. Add a test configuration for SQLite (x64). [not recommended]
-echo F. Add a test configuration for PostgreSQL.
-echo G. Add a test configuration for Oracle.
-echo H. Add a test configuration for SQL Server Compact (x86).
-echo I. Add a test configuration for SQL Server Compact (x64).
-echo J. Add a test configuration for MySql.
+echo B. Add a test configuration for Firebird.
+echo C. Add a test configuration for SQLite.
+echo D. Add a test configuration for PostgreSQL.
+echo E. Add a test configuration for Oracle.
+echo F. Add a test configuration for SQL Server Compact (x86).
+echo G. Add a test configuration for SQL Server Compact (x64).
+echo H. Add a test configuration for MySql.
 echo.
 echo X.  Exit to main menu.
 echo.
 
-%BUILDTOOL% prompt ABCDEFGHIJX
-if errorlevel 10 goto main-menu
-if errorlevel 9 goto test-setup-mysql
-if errorlevel 8 goto test-setup-sqlservercex64
-if errorlevel 7 goto test-setup-sqlservercex86
-if errorlevel 6 goto test-setup-oracle
-if errorlevel 5 goto test-setup-postgresql
-if errorlevel 4 goto test-setup-sqlitex64
-if errorlevel 3 goto test-setup-sqlitex86
-if errorlevel 2 goto test-setup-firebirdx64
-if errorlevel 1 goto test-setup-firebirdx86
+%BUILDTOOL% prompt ABCDEFGHX
+if errorlevel 8 goto main-menu
+if errorlevel 7 goto test-setup-mysql
+if errorlevel 6 goto test-setup-sqlservercex64
+if errorlevel 5 goto test-setup-sqlservercex86
+if errorlevel 4 goto test-setup-oracle
+if errorlevel 3 goto test-setup-postgresql
+if errorlevel 2 goto test-setup-sqlite
+if errorlevel 1 goto test-setup-firebird
 if errorlevel 0 goto test-setup-sqlserver
 
 :test-setup-sqlserver
 set CONFIG_NAME=MSSQL
-set PLATFORM=AnyCPU
+set TEST_PLATFORM=AnyCPU
 set LIB_FILES=
 set LIB_FILES2=
 goto test-setup-generic
 
 :test-setup-sqlservercex86
 set CONFIG_NAME=SqlServerCe32
-set PLATFORM=AnyCPU
+set TEST_PLATFORM=AnyCPU
 set LIB_FILES=lib\teamcity\SqlServerCe\*.dll
 set LIB_FILES2=lib\teamcity\SqlServerCe\X86\*.dll
```

Comment:
You can remove these lines for the SQL Server Compact native files by linking to the files from the test .csproj, and copying to output, and also including a PackageReference to Microsoft.SqlServer.Compact.

Alternatively, a custom `.target` file can be used to pull the files from the `$(NuGetPackageRoot)` and copy them to output.

Reply:
Never mind about the `.target`, that's just how I automate the whole thing for subsequent dependencies.  For just a single test project, include this:

```xml
  <ItemGroup>
    <PackageReference Include="Microsoft.SqlServer.Compact" Version="4.0.8876.1" />
  </ItemGroup>
  <ItemGroup '$(NuGetPackageRoot)' != '' ">
      <NativeBinaries Include="$(NuGetPackageRoot)microsoft.sqlserver.compact\4.0.8876.1\NativeBinaries\**\*.*" />
      <Content Include="@(NativeBinaries)">
        <Link>%(RecursiveDir)%(Filename)%(Extension)</Link>
        <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
      </Content>
  </ItemGroup>
```

Reply:
Thanks,

Added to NHibernate.Test.csproj, seems taken into account by NHibernate.TestDatabaseSetup.csproj too, so it should fix last CE failure.